src/frontend/packages/core/src/core/entity-catalogue/entity-catalogue-entity.ts

```diff
@@ -263,7 +254,9 @@ export class StratosCatalogueEndpointEntity extends StratosBaseCatalogueEntity<I
   public definition: IStratosEndpointDefinition<EntityCatalogueSchemas>;
   constructor(
     entity: StratosEndpointExtensionDefinition | IStratosEndpointDefinition,
-    getLink?: (metadata: IEndpointFavMetadata) => string
+    getLink?: (metadata: IEndpointFavMetadata) => string,
+    // TODO: attach to PreviewableComponent
```

Comment:
TODO

Reply:
I didn't find a way to do this. We are returning a class reference and not an instance of `PreviewableComponent`. So in theory what we need was something like `Function<PreviewableComponent>`, but that doesn't work. :sweat_smile:

Reply:
You should be able to set `Type<PreviewableComponent>` as the return type for the function.